src/liberica.ts

```diff
@@ -0,0 +1,119 @@
+  javaVersion: string,
+  javaPackage: string
+): Promise<string> {
+  const resolvedJavaVersion = await findLatestLibericaJavaVersion(javaVersion)
+  const downloadUrl = await findLibericaURL(resolvedJavaVersion, javaPackage)
+  const toolName = determineToolName(javaVersion, javaPackage)
+  return downloadExtractAndCacheJDK(
+    async () => downloadTool(downloadUrl),
+    toolName,
+    javaVersion
+  )
+}
+
+export async function findLatestLibericaJavaVersion(
+  javaVersion: string
+): Promise<string> {
+  const matchingRefs = await getMatchingTags(
+    LIBERICA_GH_USER,
+    LIBERICA_RELEASES_REPO,
+    `${LIBERICA_JDK_TAG_PREFIX}${javaVersion}`
+  )
+  const noMatch = '0.0.1'
+  let bestMatch = noMatch
+  const prefixLength = `refs/tags/${LIBERICA_JDK_TAG_PREFIX}`.length
+  const patternLength = javaVersion.length
+  for (const matchingRef of matchingRefs) {
+    const version = matchingRef.ref.substring(prefixLength)
+    if (
+      semver.valid(version) &&
+      // pattern '17.0.1' should match '17.0.1+12' but not '17.0.10'
+      (version.length <= patternLength ||
+        !isDigit(version.charAt(patternLength))) &&
+      semver.compareBuild(version, bestMatch) == 1
+    ) {
+      bestMatch = version
+    }
+  }
+  if (bestMatch === noMatch) {
+    throw new Error(
+      `Unable to find the latest version for JDK${javaVersion}. Please make sure the java-version is set correctly. ${c.ERROR_HINT}`
+    )
+  }
+  return bestMatch
+}
+
+export async function findLibericaURL(
+  javaVersion: string,
+  javaPackage: string
+): Promise<string> {
+  const release = await getTaggedRelease(
+    LIBERICA_GH_USER,
+    LIBERICA_RELEASES_REPO,
+    LIBERICA_JDK_TAG_PREFIX + javaVersion
+  )
+  const platform = determinePlatformPart()
+  const assetPrefix = `${LIBERICA_VM_PREFIX}${determineVariantPart(
+    javaPackage
+  )}openjdk${javaVersion}`
+  const assetSuffix = `-${platform}${c.GRAALVM_FILE_EXTENSION}`
+  for (const asset of release.assets) {
+    if (
+      asset.name.startsWith(assetPrefix) &&
+      asset.name.endsWith(assetSuffix)
+    ) {
+      return asset.browser_download_url
+    }
+  }
+  throw new Error(
+    `Unable to find asset for java-version: ${javaVersion}, java-package: ${javaPackage}, platform: ${platform}`
+  )
+}
+
+function determineToolName(javaVersion: string, javaPackage: string) {
+  const variant = determineVariantPart(javaPackage)
+  const platform = determinePlatformPart()
+  return `${LIBERICA_VM_PREFIX}${variant}openjdk${javaVersion}-${platform}`
+}
+
+function determineVariantPart(javaPackage: string) {
+  return javaPackage !== null && javaPackage.includes('+fx') ? 'full-' : ''
+}
+
+function determinePlatformPart() {
+  if (isMuslBasedLinux()) {
+    return `linux-${c.JDK_ARCH}-musl`
+  } else {
+    return `${c.JDK_PLATFORM}-${c.GRAALVM_ARCH}`
+  }
+}
+
+function isMuslBasedLinux() {
+  if (c.IS_LINUX) {
+    const output = spawnSync('ldd', ['--version']).stderr.toString('utf8')
```

Comment:
So you'll automatically switch to a musl version of Liberica on musl-based Linux system. Are you sure GitHub Action can provide musl-based systems? If yes, how, via Docker? Even though `setup-java` [mentions `linux-musl`](https://github.com/actions/setup-java/blob/99b8673ff64fbf99d8d325f52d9a5bdedb8483e9/src/distributions/liberica/installer.ts#L19), it does not seems to be [actually supported](https://github.com/actions/setup-java/blob/99b8673ff64fbf99d8d325f52d9a5bdedb8483e9/src/distributions/liberica/installer.ts#L159-L172).

Maybe this additional logic can be safely removed to keep things simple?

Reply:
I believe they use Docker images, and do have at least [some support](https://github.com/actions/runner-images/pull/7274) for Alpine. Plus, people can set up their own self-hosted runners.

Reply:
Just wanted to point out this is feature isn't really tested anywhere.

Reply:
True; I've done some manual testing inside a container, but I'm not sure how to automate that